### PR TITLE
BUG: fixed improper rotations in spatial.transform.rotation.match_vectors

### PR DESCRIPTION
**Pull request by pmla**
#### Reference issue

Closes issue #10444

#### What does this implement/fix?

Corrects improper rotations as done in the Kabsch algorithm.

#### Additional information

A unit test has been added using the data provided by @sinanus in issue #10444

### CODE REVIEW

File: scipy/spatial/transform/rotation.py [hunk 1]
```diff
@@ -1620,6 +1620,13 @@ def match_vectors(cls, a, b, weights=None, normalized=False):
 
         B = np.einsum('ji,jk->ik', weights[:, None] * a, b)
         u, s, vh = np.linalg.svd(B)
+
+        # Correct improper rotation if necessary (as in Kabsch algorithm)
+        d = np.linalg.det(u @ vh) < 0
```
Comment (tylerjereddy) [inline scipy/spatial/transform/rotation.py]: Minor point, but any reason not to forego the variable and just do:
`if np.linalg.det(u @ vh) < 0: `

**Review by tylerjereddy — COMMENTED** [review]
CI failures seem unrelated

May be good if @nmayorov takes a look

File: scipy/spatial/transform/rotation.py (hunk 1 above)
Comment (pmla) [reply scipy/spatial/transform/rotation.py]: Good idea.  I have made the change.

**Comment by nmayorov** [comment]
@pmla 

Checked the code and the papers. This change is correct and absolutely necessary. We somehow overlooked  that in the reference [1] he states that U and V must be chosen such that they have determinants of +1.

Great job. Build failures are unrelated, so I suggest that we should merge it.

**Comment by larsoner** [comment]
Thanks @pmla

**Comment by ilayn** [comment]
I am just basing this on @nmayorov's comment but the current implementation checks the determinant of the product not individual determinants. Hence `-I` for u and v satisfy the code but fail the theoretical condition.
If it is OK just ignore my comment please 😃

**Comment by pmla** [comment]
@ilayn the Wikipedia article on the Kabsch algorithm does a good job of describing why the determinant of the product is correct: https://en.wikipedia.org/wiki/Kabsch_algorithm

**Pull request merged**